tests/boots/test_solved_software_environment.py

```diff
@@ -59,9 +61,13 @@ def test_run(self, context: Context) -> None:
         context.project.runtime_environment.operating_system.name = "fedora"
         context.project.runtime_environment.operating_system.version = "32"
         context.project.runtime_environment.python_version = "3.8"
-        context.graph.should_receive("solved_software_environment_exists").with_args(
-            os_name="fedora", os_version="32", python_version="3.8"
-        ).and_return(True).once()
```

Comment:
The test suite is failing because of this removal. The query is still used in the pipeline unit code, which causes that the adapter is trying to be used (but there is no database to connect to) - here and below.